addons/mail/models/mail_message.py

```diff
@@ -1208,10 +1208,34 @@ def _bus_notification_target(self):
     def _cleanup_side_records(self):
         """ Clean related data: notifications, stars, ... to avoid lingering
         notifications / unreachable counters with void messages notably. """
+        outdated_starred_partners = self.starred_partner_ids
         self.write({
             'starred_partner_ids': [(5, 0, 0)],
             'notification_ids': [(5, 0, 0)],
         })
+        if outdated_starred_partners:
+            # sudo: bus.bus: reading non-sensitive last id
+            bus_last_id = self.env["bus.bus"].sudo()._bus_last_id()
+            star_count_by_partner = dict(
+                self.env["mail.message"]._read_group(
+                    [("starred_partner_ids", "in", outdated_starred_partners.ids)],
+                    groupby=["starred_partner_ids"],
+                    aggregates=["__count"],
+                )
+            )
+            notifications = []
+            for partner in outdated_starred_partners:
+                payload = {
+                    "Thread": {
+                        "id": "starred",
+                        "messages": [("DELETE", {"id": self.ids})],
```

Comment:
This should be a single id.
Also method is not flagged as `ensure_one` which makes it confusing (the only caller is ensure one though).